src/libraries/Common/src/Interop/Windows/BCrypt/Interop.BCryptImportKey.cs

```diff
@@ -16,24 +16,34 @@ internal static unsafe SafeKeyHandle BCryptImportKey(SafeAlgorithmHandle hAlg, R
             const string BCRYPT_KEY_DATA_BLOB = "KeyDataBlob";
             int keySize = key.Length;
             int blobSize = sizeof(BCRYPT_KEY_DATA_BLOB_HEADER) + keySize;
-            byte[] blob = new byte[blobSize];
+
+            // 64 is large enough to hold a BCRYPT_KEY_DATA_BLOB_HEADER and an AES-256 key with room to spare.
+            int MaxStackKeyBlob = 64;
+            Span<byte> blob = stackalloc byte[MaxStackKeyBlob];
+
+            if (blobSize > MaxStackKeyBlob)
+            {
+                blob = new byte[blobSize];
+            }
```

Comment:
We happen to be writing all three fields of the blob header, but we're in code that doesn't guarantee it.  Arrays are guaranteed zero-filled.  We should be paranoid and do the same for our stack-span.

```suggestion
            }
            else
            {
                blob.Clear();
            }
```

Alternatively, I might be willing to accept a Debug.Assert that sizeof(BCRYPT_KEY_DATA_BLOB_HEADER) == 3 * sizeof(uint) and a comment that says we can avoid zero-filling the stackalloc ... right next to where we assign the three fields.

Reply:
The difference between non-clearing and clearing is negligible when I benchmarked it - there are bigger fish to fry. So clearing it is.